scripts/ci/ws12_workflow_contracts.py

```
@@ -0,0 +1,97 @@
+
+def validate_workflow_texts(workflows: dict[str, str]) -> list[str]:
+    """Return every missing lane marker; an empty result is the only pass."""
+    errors: list[str] = []
+    for path, markers in REQUIRED_MARKERS.items():
+        text = workflows.get(path)
+        if text is None:
+            errors.append(f"missing workflow: {path}")
+            continue
+        for marker in markers:
+            if marker not in text:
+                errors.append(f"{path}: missing {marker!r}")
```

Ruff (0.16.0)

[warning] Use `list.extend` to create a transformed list

(PERF401)